src/components/listings/ImageGridItem.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from 'react'
+import { ImageBackground, Pressable } from 'react-native'
+import Animated, {
+  useSharedValue,
+  withDelay,
+  withTiming,
+} from 'react-native-reanimated'
+import Icon from '@/components/Icon'
+import { useColours } from '@/hooks/useTailwind'
+import { IconSize } from '@/types/icon'
+import { ImageGridItemProps } from '@/types/imageGrid'
+
+const DURATION = 200
+const DELAY = 50
+
+function ImageGridItem(props: ImageGridItemProps) {
+  const { image, setSelectedImages, selectedImages } = props
+
+  const colours = useColours()
+
+  const [selected, setSelected] = useState(
+    selectedImages.includes(image.id) || false,
```

Comment:
do we need to pass the whole array to do this? can't we do this in the image grid component instead?

Reply:
👍